docs/typescript.md

```diff
@@ -1,29 +1,76 @@
 ---
 layout: default
-title: '[EXPERIMENTAL] Using TypeScript with DNSControl'
+title: Using TypeScript with DNSControl
 ---
 
-# [EXPERIMENTAL] Using TypeScript with DNSControl
+# Using TypeScript with DNSControl (experimental)
 
-> **NOTE**: This feature is currently experimental. Type information may change in future releases, and any release (even a patch release) could introduce new type-checking errors into your `dnsconfig.js` file. If you plan to use TypeScript, please file bug reports for any issues you encounter, and avoid manually running type-checking as part of your deployment process until the “experimental” label is removed.
+## What is this?
 
-While DNSControl does not support TypeScript syntax in `dnsconfig.js`, you can still use TypeScript’s features in editors which support it (including Visual Studio Code). To set up TypeScript support in Visual Studio Code, follow these steps:
-First, you’ll need to grab the `dnscontrol.d.ts` file corresponding to your version of DNSControl. You can manually download this file off of GitHub, or you can use the following command:
+Would you like your editor to support auto-completion and other advanced IDE
+features when editing dnsconfig.js?  Yes you can!
+
+While DNSControl does not support TypeScript syntax in `dnsconfig.js`, you can
+still use TypeScript’s features in editors which support it.
+
+If you’re using VS Code (or another editor that supports TypeScript), you
+should now be able to see the type information in your `dnsconfig.js` file as
+you type. Hover over record names to read their documentation without having to
+open the website!
+
+## How to activate
+
+To set up TypeScript support in Visual Studio Code, follow these steps:
+
+1. Run this command to generate the file `types-dnscontrol.d.ts`.
```

Comment:
I have not tested the TypeScript setup before, I am not getting a populated `types-dnscontrol.d.ts` file at the moment.

```shell
brew upgrade --greedy
```

```text
==> Upgrading dnscontrol
  3.23.0 -> 3.24.0

==> Pouring dnscontrol--3.24.0.ventura.bottle.tar.gz
/usr/local/Cellar/dnscontrol/3.24.0: 5 files, 35.3MB
==> Running `brew cleanup dnscontrol`...
Removing: /usr/local/Cellar/dnscontrol/3.23.0... (5 files, 35.4MB)
```

```shell
dnscontrol write-types
```
```text
Downloading https://raw.githubusercontent.com/StackExchange/dnscontrol//types/dnscontrol.d.ts to types-dnscontrol.d.ts... done.
```

```shell
dnscontrol version
```
```text
dnscontrol -dev
```

In response to this, I opened https://github.com/StackExchange/dnscontrol/issues/1943.

Reply:
Copy/pasting the content from https://github.com/StackExchange/dnscontrol/blob/v3.24.0/types/dnscontrol.d.ts does work in JetBrains IDE (PhpStorm/GoLand)

Reply:
I just noticed that too (https://github.com/StackExchange/dnscontrol/pull/1942#issuecomment-1382423860), should hopefully be fixed for future releases by that PR though.